src/test/groovy/com/google/protobuf/gradle/UtilsTest.groovy

```diff
@@ -0,0 +1,66 @@
+package com.google.protobuf.gradle
+
+import com.android.builder.model.Version
+import org.gradle.api.Project
+import org.gradle.testfixtures.ProjectBuilder
+import spock.lang.Requires
+import spock.lang.Specification
+
+/**
+ * Unit tests for utility functions
+ */
+class UtilsTest extends Specification {
+
+  private Project setupBasicAndroidLibraryProject() {
+    Project project = ProjectBuilder.builder().build()
+    project.apply plugin:'com.android.library'
+    return project
+  }
+
+  private Project setupThreeZeroZeroUpAndroidProject() {
+    Project project = ProjectBuilder.builder().build()
+    project.apply plugin:'com.android.base'
+    return project
+  }
+
+  private Project setupBasicJavaProject(Project parent) {
+    Project project = ProjectBuilder.builder().withParent(parent).build()
+    project.apply plugin:'java'
+    return project
+  }
+
+  void "test Utils.isAndroidProject returns false for multi-module java child project"() {
+    given: "a multi-module project with a child java project and parent android project"
+    Project parent = setupBasicAndroidLibraryProject()
+    Project project = setupBasicJavaProject(parent)
+
+    when: "Utils.isAndroidProject examines child java project"
+    boolean isAndroidProject = Utils.isAndroidProject(project)
+
+    then: "Utils.isAndroidProject determines the child java project is not an android project"
+    assert !isAndroidProject
+  }
+
+  void "test Utils.isAndroidProject returns true for basic android library project"() {
+    given: "a basic android library project"
+    Project project = setupBasicAndroidLibraryProject()
+
+    when: "Utils.isAndroidProject examines project"
+    boolean isAndroidProject = Utils.isAndroidProject(project)
+
+    then: "Utils.isAndroidProject determines the basic android project is an android project"
+    assert isAndroidProject
+  }
+
+  @Requires({ Version.ANDROID_GRADLE_PLUGIN_VERSION.startsWith("3.") })
```

Comment:
Should this be for only 3.x or for 3.x and up?

Reply:
`project.apply plugin:'com.android.base'` will only work from 3.x and up. they add that marker plugin for this exact use case. My thought is that when you can safely abandon pre  3.0 support that case would be all that is required. which is why I have it as the first thing the isAndroidProject checks for. 

The problem though is this test will fail with a no such plugin error when the tests are built/ran with 2.+ plugin versions.

Reply:
I'm a bit concerned about the string matching here for the "and up" part of "3.x and up". Should we adapt `Utils.compareGradleVersion` to be able to check any version string?

Reply:
Yea if you think it is worth while I would be happy to make the change. Was just trying to get a guard in there fast so that if someone needed to debug against 2.3.0 they still could.